pkg/tsdb/tempo/tempo.go

```diff
@@ -0,0 +1,124 @@
+package tempo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/tsdb"
+
+	jaeger "github.com/jaegertracing/jaeger/model"
+	jaeger_json "github.com/jaegertracing/jaeger/model/converter/json"
+
+	ot_pdata "go.opentelemetry.io/collector/consumer/pdata"
+	ot_jaeger "go.opentelemetry.io/collector/translator/trace/jaeger"
+)
+
+type tempoExecutor struct {
+	httpClient *http.Client
+}
+
+func newTempoExecutor(dsInfo *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+	httpClient, err := dsInfo.GetHttpClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &tempoExecutor{
+		httpClient: httpClient,
+	}, nil
+}
+
+var (
+	plog log.Logger
+)
+
+func init() {
+	plog = log.New("tsdb.tempo")
+	tsdb.RegisterTsdbQueryEndpoint("tempo", newTempoExecutor)
+}
+
+func (e *tempoExecutor) Query(ctx context.Context, dsInfo *models.DataSource, tsdbQuery *tsdb.TsdbQuery) (*tsdb.Response, error) {
```

Comment:
Just so you're aware, we're currently trying to migrate away from the typical legacy tsdb contracts to the newer backend SDK IE
`func QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error)`

Are you aiming to get this into the 7.5 beta? If not, then would be happy to help update this.

Reply:
Good to know Will. I'd love to get this in for 7.5 but I'm also up for improving this.

Reply:
I guess considering it should be for the beta then it might be too much to change this as we're hoping to have everything merged by EOD today. It should be fairly straightforward to fix after the fact 😄